Cleanup fails with rm permission denied in go/pkg/mod (action exits 1)

## Summary
`misty-step/cerberus@17ba992dafd4849e8249fa90cd0df9d9c2d8c2d7` is failing jobs during cleanup with `rm ... Permission denied` under the Go module cache path.

This appears to be an action/runtime bug, not repo code behavior.

## Where observed
- Workflow run: https://github.com/phrazzld/glance/actions/runs/22267352535
- PR: https://github.com/phrazzld/glance/pull/59
- Failing jobs (Cerberus lanes):
  - Architecture & Design: https://github.com/phrazzld/glance/actions/runs/22267352535/job/64416291939
  - Testing & Coverage: https://github.com/phrazzld/glance/actions/runs/22267352535/job/64416291928
- Performance & Scalability on the same run was also failing with the same cleanup error initially; later rerun passed, which suggests flakiness.

## Error excerpt
```text
rm: cannot remove '/tmp/cerberus.bHA0cQ/cerberus-home-testing.nBReaT/go/pkg/mod/go.opentelemetry.io/otel@v1.35.0/...': Permission denied
##[error]Process completed with exit code 1.
```

## Why this looks like Cerberus/action-level
- The branch is docs-only and local checks pass.
- Failure happens in cleanup phase (mass `rm` under `/tmp/cerberus.../go/pkg/mod/...`), not in review logic.
- Re-running failed jobs (`gh run rerun --failed`) did not reliably clear it.

## Expected behavior
- Cleanup should not fail due to file ownership/permissions inside the action workspace, or at least should not hard-fail the job after successful analysis.

## Possible fix directions
- Normalize permissions before cleanup (`chmod/chown` defensively on the temp `go/pkg/mod` tree).
- Avoid creating root-owned files in the temp HOME/cache.
- Make cleanup best-effort (`rm -rf ... || true`) after primary review steps complete.

--- a/scripts/run-reviewer.sh
+++ b/scripts/run-reviewer.sh
@@ -28,6 +28,9 @@ cerberus_cleanup() {
   # Note: model-used is NOT cleaned here — downstream steps read it.
   # Note: parse-failure-models.txt and parse-failure-retries.txt are NOT
   # cleaned here — parse-review.py reads them to enrich SKIP verdicts.
+  # Go module cache (go/pkg/mod) may have restrictive permissions; chmod first
+  # to ensure we can delete, then remove. Both operations are best-effort.
+  chmod -R 755 "${CERBERUS_ISOLATED_HOME:-}" 2>/dev/null || true
   rm -rf "${CERBERUS_ISOLATED_HOME:-}" 2>/dev/null || true
 
   if [[ -z "$cerberus_staging_backup_dir" ]]; then

